jupyter_elfinder/elfinder.py

```diff
@@ -1439,30 +1439,31 @@ def __put(self) -> None:
 
     def __archive(self) -> None:
         """Compress files/directories to archive."""
-        if (
-            not self._options["archivers"]["create"]
-            or API_TYPE not in self._request
-            or API_TARGET not in self._request
-            or API_TARGETS not in self._request
-        ):
+        # TODO: We don't support "name" field yet.
+        # "name" is a parameter according to api 2.1.
+        archive_type = self._request.get(API_TYPE)
+        target = self._request.get(API_TARGET)
+        files = self._request.get(API_TARGETS)
+        if not archive_type or not target or not files:
             self._response[R_ERROR] = "Invalid parameters"
             return
 
-        cur_dir = self.__find_dir(self._request[API_TARGET])
-        archive_type = self._request[API_TYPE]
+        cur_dir = self.__find_dir(target)
+        if not cur_dir:
+            self._response[R_ERROR] = "File not found"
+            return
+
+        if not self.__is_allowed(cur_dir, "write"):
+            self._response[R_ERROR] = "Access denied"
+            return
+
         if (
             archive_type not in self._options["archivers"]["create"]
             or archive_type not in self._options["archiveMimes"]
-            or not cur_dir
-            or not self.__is_allowed(cur_dir, "write")
         ):
             self._response[R_ERROR] = "Unable to create archive"
             return
 
-        files = self._request[API_TARGETS]
-        if not isinstance(files, list):
```

Comment:
This can't happen since targets is always a list:

https://github.com/imjoy-team/jupyter-elfinder/blob/d52dafb8bcf9d0227f9a76f2066173a57b408ea3/jupyter_elfinder/views.py#L100-L101